### ISSUE TITLE
Repeater styling

### ISSUE THREAD

**Issue by mrleblanc101** [laravel/nova-issues#6166]
- Laravel Version: 10.41.0
- Nova Version: 4.32.11
- PHP Version: 8.1

### Description:
Is it normal that the repeater trash icon is not uniform with the rest of nova ?

<img width="1139" alt="Screenshot 2024-01-25 at 12 36 31 AM" src="https://github.com/laravel/nova-issues/assets/17392251/4dc2f3f7-cd1f-47d5-8344-294d6bdb2a40">

<img width="169" alt="Screenshot 2024-01-25 at 12 42 08 AM" src="https://github.com/laravel/nova-issues/assets/17392251/329c42f4-cfc7-46f5-ad33-41a17f758c20">

**Comment by mrleblanc101** [comment]
Also, my field is show in the edit view and save just fine.

<img width="1800" alt="Screenshot 2024-01-25 at 12 44 59 AM" src="https://github.com/laravel/nova-issues/assets/17392251/32828fab-64c7-4eb8-a742-968dd8390fe3">

But is it normal that my field is not shown in the detail view ?

<img width="1800" alt="Screenshot 2024-01-25 at 12 45 07 AM" src="https://github.com/laravel/nova-issues/assets/17392251/a70aabd3-f2b8-4bee-8c5d-bb0b419a00cc">

**Comment by mrleblanc101** [comment]
The `->confirmRemoval()` does not appear to do anything either 🤔

**Comment by Navi2016** [comment]
Having the same, seems weird it does not show anything when viewing the resource detail page. It should display an overview of the line items.

**Comment by mrleblanc101** [comment]
Not stale

**Comment by sp4cecat** [comment]
> @Navi2016 Yeah I opened a PR to the docs, but they don't seem to care: [laravel/nova-docs#570](https://github.com/laravel/nova-docs/pull/570).

I don't think it's a matter of not caring - it's not apparent from that pull request that you're wanting the repeater details to display in the detail view.

**Comment by sp4cecat** [comment]
Which BTW I would like to see also

**Comment by caongan2** [comment]
`Text::make('Item content', 'line_items', function ($value, $resource, $attribute) {
            if ($value) {
                $value = json_decode($value);
                $arr = [];
                foreach ($value as $item) {
                    $arr[] = $item->fields->item_content;
                }

                return implode("<br>", $arr);
            }
        })->asHtml();`

**Comment by sp4cecat** [comment]
> Text::make('Item content', 'line_items', function ($value, $resource, $attribute) { if ($value) { $value = json_decode($value); $arr = []; foreach ($value as $item) { $arr[] = $item->fields->item_content; } return implode("", $arr); } })->asHtml();

Thanks for having a crack :)

jsondecode is not needed if you've cast the repeated field in your model file, ie:

`	protected $casts = [
		'line_items' => 'array',
	];
`
And also the individual line item structure is a bit more complex than that - 

- type (string)
- fields (array)
  \
    [key, value]...

**Comment by christian-heiko** [comment]
Any Update/Roadmap on this?

**Comment by christian-heiko** [comment]
Ive made a quick Shortcut where one can pass a Repeater to view it on Detail:

**Usage:**
```php
Panel::make('Panel', [
     ...RepeaterDetailView::makeBoth(Repeater::make('Label', 'attribute'), $request)
]);
```

**app/Nova/Fields/RepeaterDetailView.php**
```php
<?php

namespace App\Nova\Fields;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Collection;
use Laravel\Nova\Fields\Field;
use Laravel\Nova\Fields\Hidden;
use Laravel\Nova\Fields\Repeater;
use Laravel\Nova\Fields\Text;
use Laravel\Nova\Http\Requests\NovaRequest;

class RepeaterDetailView {

    public static function makeBoth(Repeater $repeater, NovaRequest $request): array {
        return [
            $repeater,
            self::make($repeater, $request)
        ];
    }

    public static function make(Repeater $repeater, NovaRequest $request): Text {
        return Text::make(
            $repeater->name, function (Model $model) use ($repeater, $request): string {
                $items = $model->{$repeater->attribute};

                if (is_null($items)) return '';

                $tables = [];

                foreach ($items as $item) {
                    $type = $repeater->repeatables->findByKey($item['type']);
                    /** @var Collection<Field> $fields */
                    $fields = $type->fields($request);
                    $table = [
                        'headers' => [],
                        'rows' => [[]]
                    ];

                    foreach ($fields as $field) {
                        if ($field instanceof Hidden) continue;

                        $table['headers'][] = $field->name;
                        $table['rows'][0][] = $item['fields'][$field->attribute];
                    }
                    $tables[] = $table;
                }

                return implode(
                    '<br />',
                    array_map(
                        fn(array $table) => view('partials.table',
                            $table
                        )->render(),
                        $tables
                    )
                );
        })->onlyOnDetail()->asHtml();
    }

}

```

**table.partials.php**
```blade
<table class="table w-full divide-y divide-gray-100 dark:divide-gray-700 mb-6">
    <thead>
    <tr>
        @foreach($headers as $header)
            <th>{{$header}}</th>
        @endforeach
    </tr>
    </thead>
    <tbody class="divide-y divide-gray-100 dark:divide-gray-700">
    @foreach($rows as $row)
        <tr class="table-row">
            @foreach($row as $cell)
                <td class="">
                    {{ $cell }}
                </td>
            @endforeach
        </tr>
    @endforeach
    </tbody>
</table>
```

**Comment by bergstar** [comment]
This wasn't obvious for me, so posting for other who might strangle with this issue.

Basically in your Nova/Resource file you can have both Repeater and HasMany:

```
            Repeater::make('Line Items')
                ->repeatables([
                    \App\Nova\Repeater\LineItem::make()
                ])
                ->asHasMany()
                ->showOnDetail(),

            HasMany::make('Line Items'),
```

However for some unknown reason, each calls a different functions in Model:

`Repeater::make('Line Items') calls line_items()`

but

`HasMany::make('Line Items') calls lineItems()`

You can however mitigate that by a few more arguments:

`HasMany::make('Line Items', 'line_items', \App\Nova\LineItem::class),`

This way you will see Repeater editor while editing and table view of when viewing. 

<img width="943" alt="Screenshot 2024-05-17 at 22 30 02" src="https://github.com/laravel/nova-issues/assets/1925892/df9a4f52-e34c-40c1-82cd-e19a79a924fd">

**Comment by shriramms99** [comment]
I would also like to see the JSON column details on the resource page as mentioned in https://github.com/laravel/nova-issues/issues/6166#issuecomment-1909396838 by @mrleblanc101

**Comment by msucevan** [comment]
@crynobone thank you for closing my issue without a comment!

**Comment by msucevan** [comment]
@crynobone should you move also this issue into a discussion or only my issues MUST to be moved into discussion without an answer?

**Comment by msucevan** [comment]
+1

**Comment by PaLemmen** [comment]
Not totally sure if I just stack below this issue, but the name of the issue is 'Repeater styling' so here I go.
I am trying to use the Repeater field as a block builder, but the select when having multiple repeaterables isn't that UX friendly.

<img width="251" alt="Screenshot 2024-08-12 at 23 20 01" src="https://github.com/user-attachments/assets/32fbfb91-0c8a-41f1-85dd-5770d1c53f32">

When having multiple types of repeatables the vue swaps to Dropdownmenu instead of InvertedButton.
Only the DropdownMenu inside of the RepeaterField.vue is missing a width defination or the fallback width value of the DropdownMenu should be 'auto'.

In the current situation when nothing is defined it will be fixed to 120px.
Which isn't a lot with also the icon before, and leaving therefor only space for about 7 character as label.

**Comment by crynobone** [comment]
Released with Laravel Nova 5.0.0

Feel free to open up a new issue if you're still experiencing this problem on the latest version.
